sklearn/model_selection/_search.py

```diff
@@ -621,14 +621,31 @@ def fit(self, X, y=None, groups=None, **fit_params):
                     not isinstance(self.refit, six.string_types) or
                     # This will work for both dict / list (tuple)
                     self.refit not in scorers):
-                raise ValueError("For multi-metric scoring, the parameter "
-                                 "refit must be set to a scorer key "
-                                 "to refit an estimator with the best "
-                                 "parameter setting on the whole data and "
-                                 "make the best_* attributes "
-                                 "available for that metric. If this is not "
-                                 "needed, refit should be set to False "
-                                 "explicitly. %r was passed." % self.refit)
+                if callable(self.refit):
+                    scorer_key = self.refit.__name__.split('_')[-1]
+                    if scorer_key not in scorers:
+                        raise ValueError("For multi-metric scoring, the "
+                                         "name of refit callable function "
+                                         "must end with a scorer "
+                                         "key('_<scorer_name>') to refit an "
+                                         "estimator with the best parameter "
+                                         "setting on the whole data and make "
+                                         "the best_* attributes available for "
+                                         "that metric. If this is not needed, "
+                                         "refit should be set to False "
+                                         "explicitly. %r was passed"
+                                         % self.refit)
+                    refit_metric = scorer_key
```

Comment:
I don't get why you need this. you don't use `refit_metric` if refit is callable below. I also think making inferences from the name of the function is inappropriate.

Reply:
I think `refit_metric` is needed to compute self.best_score_ as shown [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_search.py#L764) in the original code base.

Reply:
Ah, I see. I would just disable `best_score_` when refit is callable. Please test and document that behaviour.